tests/unit/model-test.js

```diff
@@ -713,6 +713,52 @@ module('unit/model', function(hooks) {
     assert.equal(get(model, 'hb'), true, 'alias to missing with default');
   });
 
+  test('.unknownProperty resolves id-matched values to external m3-models when special ref present in _data passed to schema', function(
+    assert
+  ) {
+    let model = run(() => {
+      return this.store.push({
+        data: {
+          id: 'isbn:9780439708180',
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            name: `Harry Potter and the Sorcerer's Stone`,
+            pubDate: 'September 1989',
+            '*relatedBook': 'isbn:9780439136365',
+          },
+        },
+        included: [
+          {
+            id: 'isbn:9780439136365',
+            type: 'com.example.bookstore.Book',
+            attributes: {
+              name: `Harry Potter and the Order of the Phoenix`,
+            },
+          },
+        ],
+      });
+    });
+
+    SchemaManager.schema.computeAttributeReference = function(key, value, modelName, data) {
+      let refValue = data['*' + key];
+      if (typeof value === 'string' && refValue && refValue === value && /^isbn:/.test(value)) {
+        return {
+          type: null,
+          id: value,
+        };
+      }
+      return null;
+    };
+
+    model.set('relatedBook', 'isbn:9780439136365');
```

Comment:
This seems to be testing that `computeAttributeReference` has a priority over a preexisting value, e.g. `computeAttributeReference` will always be consulted before using the value as is.

I generally prefer to keep the tests (especially unit tests) focused on what they are testing and nothing more.

Reply:
Removed `set` from the test